src/hotspot/share/cds/archiveBuilder.hpp

```diff
@@ -179,8 +179,8 @@ class ArchiveBuilder : public StackObj {
 
   class SrcObjTableCleaner {
   public:
-    bool do_entry(address key, const SourceObjInfo* value) {
-      delete value->ref();
+    bool do_entry(address key, const SourceObjInfo& value) {
+      delete value.ref();
```

Comment:
One of the things that I thought that should change with the _src_obj_table is that SourceObjInfo should have a destructor that should call this delete.  But then the table would have to take SourceObjInfo elements and not copy from a stack object.  I think ResourceHashtable's destructor calls delete on all the nodes.  Anyway, that can be a future improvement if you've followed what I mean.

Reply:
Yes, I plan to add calls to the elements' destructors (perhaps both ~K() and ~V()?) in a separate PR. This would be similar to what `GrowableArray` does when the array elements are removed, or when the array is destroyed.

To do this, we need to get `SourceObjInfo` to work with C++ Move Semantics so that the `_ref` field will not be freed more than once. I am still trying to learn how that works ....